app/lib/dttp/batch_request.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module Dttp
+  class BatchRequest
+    class Error < StandardError; end
+
+    attr_reader :batch_id, :change_set_id, :headers, :change_sets
+
+    def initialize(batch_id: SecureRandom.uuid, change_set_id: SecureRandom.uuid)
```

Comment:
Convenience is one thing, but is there a need to store `uuid` for auditing/accountability as we well as for replays?

Reply:
As far as I know, we don't need to store the `batch_id` or `change_set_id` for auditing. It's just a requirement from the Web API in order to send a batch request. You actually don't even need to use a `uuid` - it can be any unique identifier.

With regards to replays, I don't think that's on the cards right now. The batch request either succeeds or fails. We're going to have a background job for this and if it fails, it will retry 5 times before alerting Sentry.